### PR TITLE
Compile Caching

### PR DESCRIPTION
**Pull request by bircni**
Added compile caching with sccache to github workflows to make them faster

* [x] I have followed the instructions in the PR template

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -124,6 +134,10 @@ jobs:
 
       - name: Set up cargo cache
         uses: Swatinem/rust-cache@v2
+
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.6
+
```
Comment (lucasmerlin) [inline .github/workflows/rust.yml:140]: Does it make sense to use two different build caches?

**Review by lucasmerlin — COMMENTED** [review]
How much does the pipeline time decrease with these changes?

**Review by emilk — COMMENTED** [review]
We already have https://github.com/Swatinem/rust-cache set up for the dependencies - how is this different?

**Comment by bircni** [comment]
When using local cache for local builds this could improve build times massively otherwise here it seems like we unfortunately dont have any improvements -> will have a look at how to improve local compile caching!

**Pull request closed without merging**